invenio_app_ils/document_requests/views.py

```diff
@@ -88,70 +95,163 @@ def __init__(self, serializers, ctx, *args, **kwargs):
             setattr(self, key, value)
 
 
-class PendingRequestResource(DocumentRequestActionResource):
-    """Pending document request action resource."""
+class DocumentRequestDocumentResource(DocumentRequestActionResource):
+    """Endpoint to manage the Document of the Document Request."""
 
-    view_name = "{}_pending_request"
+    view_name = "{}_document"
 
     @pass_record
-    @need_permissions("document-request-pending")
+    @need_permissions("document-request-accept")
     def post(self, pid, record, **kwargs):
-        """Pending request post method."""
-        # expecting remove_fields list of fields to remove
+        """Add or replace a Document in Document Request."""
         data = self.loader()
-        remove_fields = data.get("remove_fields", [])
 
-        for field in remove_fields:
-            del record[field]
+        if record["state"] != "PENDING":
+            raise DocumentRequestError(
+                "You cannot add Document when the Document Request "
+                "is in state: {}".format(record["state"])
+            )
+
+        document_pid = data.get("document_pid")
+        if not document_pid:
+            raise DocumentRequestError("document_pid is required")
+
+        record["document_pid"] = document_pid
+        record.commit()
+        db.session.commit()
+        current_app_ils.document_request_indexer.index(record)
+        return self.make_response(pid, record, 202)
```

Comment:
I was wondering why we return `202`  as status code for all these ones and not `200`. Is there any particular reason?

Reply:
I didn't touch that part, the 202 is used in general when the request has been accepted for processing, but its not done yet if I am not wrong. I have seen it in other places too since the very beginning. I can always switch them to `200` but it would be better to talk with the team first I guess maybe someone has a better idea, @zzacharo @kprzerwa wdyt?

Reply:
I think we should switch to 200 where it makes sense!

Reply:
we could define first if the ongoing indexing means for us that the request was still not fully processed or not, IMO we could use the 202 to have user notifications of type "your request is being processed, the results will be available in x seconds" which would actually be more explanatory in the end why changes did not appear immediately.